src/test/java/lambda/part3/exercise/Exercise3.java

```diff
@@ -0,0 +1,108 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+public class Exercise3 {
+
+    private static class LazyMapHelper<T, R> {
+
+        private List <R> source;
+
+        private LazyMapHelper(List<R> source){
+            this.source = source;
+        }
+
+        public static <T> LazyMapHelper<T, T> from(List<T> list) {
+            // TODO реализация
+            return new LazyMapHelper<>(list);
+        }
+
+        public List<R> force() {
+            // TODO реализация
+            throw new UnsupportedOperationException();
+        }
+
+        public <R2> LazyMapHelper<T, R2> map(Function<R, R2> mapping) {
+            // TODO реализация
+            List<R2> result = new ArrayList<>();
+            source.forEach(R->{
+                result.add(mapping.apply(R));
+            });
+            return new LazyMapHelper<>(result);
```

Comment:
This is not `Lazy`.

Reply:
Why not? In this exercise we have only LazyMapHelper.

Reply:
Because all operations must be applied only at `force()` method (or if you like to - at `getMapped()` method). Here you must just add operation to order (for example, by using `currentFunction.andThen(mapping)`